Dockerfile

```diff
@@ -16,6 +16,7 @@ SHELL ["/bin/bash", "-l", "-c"]
 
 ENV PATH=/root/.rbenv/bin:/root/.rbenv/plugins/ruby-build/bin:$PATH
 
+ARG RUBY_VERSION
```

Comment:
isn't this variable already at line 1 though?  I thought you moved it there.

Reply:
`ARG` only exists up until the first `FROM` it encounters. It needs to be re-instantiated to be passed to the subsequent build instructions.

the `ARG` at line 1 was already present before my changes but it could technically be removed since `amazonlinux:2` doesn't use it

Reply:
also, the ARG at line 19 was there before as well but i removed it because i also thought it wasn't needed. It took me an hour of debugging before i figured that out lol